### PR TITLE
Add chunk method to Interval class for fixed-duration subdivision of intervals

### PR DESCRIPTION
**Pull request by milosobral**
- Implemented the  method to subdivide intervals into fixed-duration chunks while preserving metadata.
- Added options to remove short chunks and check for overlaps.
- Included comprehensive tests for chunking functionality, including edge cases and metadata preservation.

Adresses #59

### CODE REVIEW

File: temporaldata/interval.py [hunk 1]
```diff
@@ -457,6 +457,87 @@ def split(
...
+            start=all_starts, end=all_ends, timekeys=self.timekeys(), **kwargs
+        )
+
+        if remove_short:
+            durations = result.end - result.start
+            mask = durations >= duration
+            result_dict_filtered = {}
+            for key in result.keys():
+                result_dict_filtered[key] = getattr(result, key)[mask]
+            result = Interval(**result_dict_filtered, timekeys=self.timekeys())
```
Comment (vinamarora8) [inline temporaldata/interval.py]: A cleaner way maybe:
```python
if remove_short:
    mask = (result.end - result.start) >= durations
    result = result.select_by_mask(mask)
```

**Review by vinamarora8 — CHANGES_REQUESTED** [review]
Hey @milosobral, this is a good method to add. Left some comments about implementation.

**Review by vinamarora8 — APPROVED** [review]
LGTM!
